src/inbox/mod_inbox_rdbms_async.erl

```diff
@@ -30,51 +32,118 @@
 %% Worker callbacks
 -export([request/2, aggregate/3, verify/3]).
 
+%% Cleaner gen_server callbacks
+-export([flush_user_bin/3, flush_global_bin/2]).
+-ignore_xref([flush_user_bin/3, flush_global_bin/2]).
+
 %% Initialisation
 -spec init(mongooseim:host_type(), gen_mod:module_opts()) -> ok.
 init(HostType, Opts) ->
     AsyncOpts = prepare_pool_opts(Opts),
     mod_inbox_rdbms:init(HostType, Opts),
+    prepare_deletes(HostType, Opts),
+    mongoose_commands:register(commands()),
     start_pool(HostType, AsyncOpts),
+    start_cleaner(HostType, Opts),
     ok.
 
 stop(HostType) ->
-    mongoose_async_pools:stop_pool(HostType, inbox).
+    mongoose_commands:unregister(commands()),
+    mongoose_async_pools:stop_pool(HostType, inbox),
+    stop_cleaner(HostType).
 
 prepare_pool_opts(#{async_writer := AsyncOpts}) ->
     AsyncOpts#{pool_type => aggregate,
                request_callback => fun ?MODULE:request/2,
                aggregate_callback => fun ?MODULE:aggregate/3,
                verify_callback => fun ?MODULE:verify/3}.
 
+prepare_deletes(_HostType, _Opts) ->
+    mongoose_rdbms:prepare(inbox_move_conversation_to_bin, inbox,
+                           [luser, lserver, remote_bare_jid],
+                           <<"UPDATE inbox SET box='bin'",
+                             " WHERE luser = ? AND lserver = ? AND remote_bare_jid = ?">>),
+    mongoose_rdbms:prepare(inbox_clean_global_bin, inbox, [timestamp],
+                           <<"DELETE FROM inbox WHERE box='bin' AND timestamp < ?">>),
+    mongoose_rdbms:prepare(inbox_clean_user_bin, inbox,
+                           [lserver, luser, timestamp],
+                           <<"DELETE FROM inbox WHERE",
+                             " lserver = ? AND luser = ? AND",
+                             " box='bin' AND timestamp < ?">>),
+    ok.
+
 -spec start_pool(mongooseim:host_type(), mongoose_async_pools:pool_opts()) -> term().
 start_pool(HostType, Opts) ->
     mongoose_async_pools:start_pool(HostType, inbox, Opts).
 
+%% Clean commands
```

Comment:
I have mixed feelings about this, because we are working hard now (mostly @Premwoik) to introduce the GraphQL API and decouple the commands from the modules, introducing a API modules for each category of commands. This API module is then used by both REST and GraphQL. Here the coupling is actually even worse, because it is coupled with a specific backend module... we need GraphQL commands covering the whole REST API.